chromadb/auth/__init__.py

```diff
@@ -1,449 +1,238 @@
         super().__init__(system)
+        self._ignore_auth_paths: Dict[
+            str, List[str]
+        ] = system.settings.chroma_server_auth_ignore_paths
+        self.overwrite_singleton_tenant_database_access_from_auth = (
+            system.settings.
+            chroma_overwrite_singleton_tenant_database_access_from_auth
+        )
 
     @abstractmethod
-    def inject_credentials(self, injection_context: T) -> None:
+    def authenticate_or_raise(
+        self, headers: Headers
+    ) -> UserIdentity:
         pass
 
+    def ignore_operation(self, verb: str, path: str) -> bool:
+        if (
+            path in self._ignore_auth_paths.keys()
+            and verb.upper() in self._ignore_auth_paths[path]
+        ):
+            return True
+        return False
+
+    def read_creds_or_creds_file(self) -> List[str]:
+        _creds_file = None
+        _creds = None
+
+        if self._system.settings.chroma_server_authn_credentials_file:
+            _creds_file = str(self._system.settings[
+                "chroma_server_authn_credentials_file"
+            ])
+        if self._system.settings.chroma_server_authn_credentials:
+            _creds = str(self._system.settings[
+                "chroma_server_authn_credentials"
+            ])
+        if not _creds_file and not _creds:
+            raise ValueError(
+                "No credentials file or credentials found in "
+                "[chroma_server_authn_credentials]."
+            )
+        if _creds_file and _creds:
+            raise ValueError(
+                "Both credentials file and credentials found."
+                "Please provide only one."
+            )
+        if _creds:
+            return [c for c in _creds.split("\n") if c]
+        elif _creds_file:
+            with open(_creds_file, "r") as f:
+                return f.readlines()
+        raise ValueError(
+            "Should never happen"
+        )
 
-# SERVER-SIDE Abstractions
-
-
-class ServerAuthenticationRequest(EnforceOverrides, ABC, Generic[T]):
-    @abstractmethod
-    def get_auth_info(self, auth_info_type: AuthInfoType, auth_info_id: str) -> T:
+    def singleton_tenant_database_if_applicable(
+        self, user: Optional[UserIdentity]
+    ) -> Tuple[Optional[str], Optional[str]]:
         """
-        This method should return the necessary auth info based on the type of
-        authentication (e.g. header, cookie, url) and a given id for the respective
-        auth type (e.g. name of the header, cookie, url param).
-
-        :param auth_info_type: The type of auth info to return
-        :param auth_info_id: The id of the auth info to return
-        :return: The auth info which can be specific to the implementation
+        If settings.chroma_overwrite_singleton_tenant_database_access_from_auth
+        is False, this function always returns (None, None).
+
+        If settings.chroma_overwrite_singleton_tenant_database_access_from_auth
```

Comment:
I assume downstream code is responsible on deciding what to do if the user has access to multiple tenants/databases and this returns None?